tests/unit/s2n_ecc_evp_test.c

```diff
@@ -0,0 +1,220 @@
+
+                    /* Clean up */
+                    EXPECT_SUCCESS(s2n_ecc_evp_params_free(&server_params));
+                    EXPECT_SUCCESS(s2n_ecc_evp_params_free(&client_params));
+            }
+        }
+    }
+    {
+        /* Test s2n_ecc_evp_write_params_point for all supported curves */
+        for (int i = 0; i < s2n_ecc_evp_supported_curves_list_len; i++) {
+            struct s2n_ecc_evp_params test_params = {0};
+            struct s2n_stuffer wire;
+            uint8_t legacy_form;
+
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&wire, 0));
+
+            /* Server generates a key for a given curve */
+            test_params.negotiated_curve = s2n_ecc_evp_supported_curves_list[i];
+            EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&test_params));
+            EXPECT_SUCCESS(s2n_ecc_evp_write_params_point(&test_params, &wire));
+
+            /* Verify output is of the right length */
+            uint32_t avail = s2n_stuffer_data_available(&wire);
+            EXPECT_EQUAL(avail, s2n_ecc_evp_supported_curves_list[i]->share_size);
+
+            /* Verify output starts with the known legacy form for curves secp256r1
+             * and secp384r1*/
+            if (s2n_ecc_evp_supported_curves_list[i]->iana_id == TLS_EC_CURVE_SECP_256_R1 ||
+                s2n_ecc_evp_supported_curves_list[i]->iana_id == TLS_EC_CURVE_SECP_384_R1) {
+                EXPECT_SUCCESS(s2n_stuffer_read_uint8(&wire, &legacy_form));
+                EXPECT_EQUAL(legacy_form, 4);
+            }
+
+            EXPECT_SUCCESS(s2n_ecc_evp_params_free(&test_params));
+#if MODERN_EC_SUPPORTED
+            OPENSSL_free(wire.blob.data);
+#else
+            EXPECT_SUCCESS(s2n_stuffer_free(&wire));
+#endif
+        }
+    }
+    {
+        /* TEST s2n_ecc_evp_read_params_point for all supported curves */
+        for (int i = 0; i < s2n_ecc_evp_supported_curves_list_len; i++) {
+            struct s2n_ecc_evp_params write_params = {0};
+            struct s2n_blob point_blob;
+            struct s2n_stuffer wire;
+
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&wire, 0));
+
+            /* Server generates a key for a given curve */
+            write_params.negotiated_curve = s2n_ecc_evp_supported_curves_list[i];
+            EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&write_params));
+            EXPECT_SUCCESS(s2n_ecc_evp_write_params_point(&write_params, &wire));
+
+            /* Read point back in */
+            EXPECT_SUCCESS(
+                s2n_ecc_evp_read_params_point(&wire, s2n_ecc_evp_supported_curves_list[i]->share_size, &point_blob));
+
+            /* Check that the blob looks generally correct. */
+            EXPECT_EQUAL(point_blob.size, s2n_ecc_evp_supported_curves_list[i]->share_size);
+            EXPECT_NOT_NULL(point_blob.data);
+
+            EXPECT_SUCCESS(s2n_ecc_evp_params_free(&write_params));
+#if MODERN_EC_SUPPORTED
+            OPENSSL_free(wire.blob.data);
```

Comment:
This doesn't look right. Alloc's and Free's should always be matching and be of the same "type". If users call `s2n_stuffer_growable_alloc(&wire, 0)`, then they should always call `s2n_stuffer_free(&wire)` once they're done with it. 

Can we update `s2n_ecc_evp_write_params_point()` so that the `OPENSSL_free()` call is never needed?